sztp-agent/pkg/secureagent/utils.go

```diff
@@ -94,8 +95,10 @@ func (a *Agent) doTLSRequest(input string, url string, empty bool) (*BootstrapSe
 func generateInputJSONContent() string {
 	osName := replaceQuotes(strings.Split(linesInFileContains(OS_RELEASE_FILE, "NAME"), "=")[1])
 	osVersion := replaceQuotes(strings.Split(linesInFileContains(OS_RELEASE_FILE, "VERSION"), "=")[1])
-
-	// dmidecode.Dmidecode(true))  // This is one possibility to get hw information
+	baseboard, err := ghw.Baseboard()
+	if err != nil {
+		log.Printf("Error getting baseboard info: %v", err)
```

Comment:
@alknopfler should we return an error here instead of just logging it ?

Reply:
makes sense return 2 values (value, and error), but then, in the invocation (agent.go) we should do it out of the struct to control it, and returning in newAgent the error too to be handle by the CLI in cmd package. Something like:

```
func NewAgent(bootstrapURL, serialNumber, devicePassword, devicePrivateKey, deviceEndEntityCert, bootstrapTrustAnchorCert string) (*Agent, error) {
	inputContent,err := generateInputJSONContent()
	if err!= nil{ 
	    return nil, err
	}    
	return &Agent{
		BootstrapURL:                  bootstrapURL,
		SerialNumber:                  serialNumber,
		DevicePassword:                devicePassword,
		DevicePrivateKey:              devicePrivateKey,
		DeviceEndEntityCert:           deviceEndEntityCert,
		BootstrapTrustAnchorCert:      bootstrapTrustAnchorCert,
		ContentTypeReq:                CONTENT_TYPE_YANG,
		InputJSONContent:              inputContent,
		DhcpLeaseFile:                 DHCLIENT_LEASE_FILE,
		ProgressJSON:                  ProgressJSON{},
		BootstrapServerRedirectInfo:   BootstrapServerRedirectInfo{},
		BootstrapServerOnboardingInfo: BootstrapServerOnboardingInfo{},
	}, nil
}
```
and in the cmd package with the cli the agent instantiation should handle the error creating the agent :

```
cmd := &cobra.Command{
		Use:   "daemon",
		Short: "Run the daemon command",
		RunE: func(cmd *cobra.Command, args []string) error {
			a, err := secureagent.NewAgent(bootstrapURL, serialNumber, devicePassword, devicePrivateKey, deviceEndEntityCert, bootstrapTrustAnchorCert)
                         if err != nil {
                             return err
                         }
			return a.RunCommandDaemon()
		},
	}
```

Reply:
since `HwModel` is optional, will just use empty string `""` in case of an error